Editing toolkit: use `is_` WordPress checks and clean up hooks

ramonjd opened the pull request:
### Changes proposed in this Pull Request

Hi ho!

This PR does 3 amazing things:

- Let's remove the duplicate action/filter hooks! 🕺 It should have no side-effects. Just janitorial. 
- Let's not set wpcom_public_coming_soon ito `0` by default if it's not in the site creation meta.  This is because some signup flows might NOT want to set that option at all. It should have no side-effects. Just janitorial.
- For some reason, Ecommerce AT sites have custom page types or something that doesn't pass our isset $page check, so we're checking for common WordPress page/post types. I asked here pbxlJb-xi-p2#comment-733

WOOT

### Testing instructions

#### Simple sites

1. Fire up the old sandbox
2. Get this patch up there via `yarn dev --sync`
3. Create a site over at `/start/domains?flags=coming-soon-v2`
4. Sandbox that site
5. Check that coming soon is enable on your simple site (incognito window).
6. Check the settings page `/settings/general/{your_site}?flags=coming-soon-v2`
7. (Launch site if necessary) Toggle Coming Soon mode on and off, check that it works in an incognito browser/logged-out session
8. Be kind to pets and children

#### Ecommerce

1. Create a site over at `/start/domains?flags=coming-soon-v2` 
2. Upgrade to Ecommerce
3. In your local `/apps/editing-toolkit` directory, run `yarn build`, then zip up `/apps/editing-toolkit/editing-toolkit-plugin`
4. Deactivate the existing WordPress Editing Toolkit over at `{your_site}/wp-admin/plugins.php?plugin_status=all&paged=1&s`
<img width="400" alt="Screen Shot 2020-11-20 at 8 03 32 pm" src="https://user-images.githubusercontent.com/6458278/99781783-65d74080-2b6c-11eb-96ee-2536fec2cbb9.png">
5. Now install your editing-toolkit-plugin.zip
<img width="400" alt="Screen Shot 2020-11-20 at 8 03 43 pm" src="https://user-images.githubusercontent.com/6458278/99781835-7687b680-2b6c-11eb-80b6-0a17ec5d4fbe.png">
6. Make sure to activate it
7. Check that coming soon is enable on your AT staging site (incognito window).
8. Check the settings page `/settings/general/{your_site}?flags=coming-soon-v2`
9. Launch (if you have to), then toggle Coming Soon mode. Check that it works

ramonjd committed:
Removing dupe hooks
Also not setting wpcom_public_coming_soon if it's not in the site creation meta.

ramonjd committed:
renaming method to be a little more specific

ramonjd committed:
ecommerce sites don't page global $post available for some reason so let's target the things we want to hide using is_ methods

ramonjd committed:
tagging the coming soon page with HTML comment

deBhal commented on apps/editing-toolkit/editing-toolkit-plugin/coming-soon/coming-soon.php:
```diff
@@ -33,13 +30,22 @@ function show_coming_soon_page() {
 	remove_action( 'wp_footer', 'stats_footer', 101 );
+	add_filter( 'jetpack_disable_eu_cookie_law_widget', '__return_true', 1 );
+	add_filter( 'jetpack_enable_opengraph', '__return_false', 1 );
 	wp_enqueue_style( 'recoleta-font', '//s1.wp.com/i/fonts/recoleta/css/400.min.css', array(), PLUGIN_VERSION );
```
Maybe a blank line before the style here?

ramonjd replied:
On it :)

ramonjd committed:
Update coming-soon.php

Grouped action and filter hooks. Added space between hooks and wp_enqueue_style

ramonjd committed:
Remove Whitespace

Fixing 49 | ERROR | [x] Whitespace found at end of line Squiz.WhiteSpace.SuperfluousWhitespace.EndLine

deBhal reviewed (approved):
Looks good!

For simple sites - prelaunch/coming soon/public/private all work as expected.

AT also worked well, just a a couple of small non-blocking notes:
- Mobile doesn't work yet (site content is visible), needs 561-gh-wpcomsh
- Not related to this PR (probably), but when I launched the site from the https://wordpress.com/settings/general/julesaustestcomingsoonecommerce.wordpress.com?flags=coming-soon-v2 page, the "Privacy" panel lit up for just a moment and then faded back to grey with the launch site dialog still visible. The launch worked, and refreshing fixed it, so not critical. Maybe a reducer problem, or maybe just a network hiccup?

Also non-blocking random comments:

- That `die()` behind the `should_show_coming_soon()` check on `template_redirect`  seems uncomfortably forceful. It seems to work, though, so we should probably just back away slowly.
- Explicitly casting the `wpcom_public_coming_soon` option to an `int` is surprising when it's a boolean everywhere else.

Those e2e failures are timeouts on editing posts. I'm assuming they're just intermittent, but we should re-run them before deploying, as there is a chance they'll need to be updated for coming soon

ramonjd commented:
Thanks for the detailed review @deBhal !!!

> That die() behind the should_show_coming_soon() check on template_redirect seems uncomfortably forceful. It seems to work, though, so we should probably just back away slowly.

What would work better? I can do a follow-up PR.

> Explicitly casting the wpcom_public_coming_soon option to an int is surprising when it's a boolean everywhere else.

Hmmm it should be an int in most places. I'll add this to the follow-up PR to double-check

ramonjd commented:
> Those e2e failures are timeouts on editing posts. I'm assuming they're just intermittent, but we should re-run them before deploying, as there is a chance they'll need to be updated for coming soon

Good idea. They were passing this morning, but I added a PHPCS-related commit and they started failing again. 🤷
